src/services/inplayer.account.service.ts

```diff
@@ -328,3 +333,75 @@ function parseJson(value: string, fallback = {}) {
 export const canUpdateEmail = false;
 
 export const canChangePasswordWithOldPassword = true;
+
+export const initCustomerExtras = async (): Promise<ExternalData> => {
+  const [favoritesData, historyData] = await Promise.all([InPlayer.Account.getFavorites(), await InPlayer.Account.getWatchHistory({})]);
+
+  const favorites = favoritesData.data?.collection?.map((favorite: FavoritesData) => {
+    return processFavorite(favorite);
+  });
+
+  const history = historyData.data?.collection?.map((history: WatchlistHistory) => {
+    return processHistoryItem(history);
+  });
+
+  return {
+    favorites,
+    history,
+  };
+};
+
+export const updatePersonalShelves: UpdatePersonalShelves = async (payload) => {
+  const { favorites, history } = payload.externalData;
+  const user = useAccountStore.getState().user;
+  const currentFavoriteIds = user?.externalData?.favorites?.flatMap((e) => e.mediaid);
```

Comment:
Do these need to be flatmap or just map?

Reply:
it can be just `map` I think

Reply:
done